code/src/main/kotlin/com/expediagroup/sdk/core/http/Response.kt

```diff
@@ -155,7 +137,7 @@ class Response private constructor(
          * @throws IllegalArgumentException If [name] or [value] is invalid.
          */
         fun addHeader(name: String, value: String) = apply {
-            headers.add(name, value)
+            headersBuilder.add(name, value)
```

Comment:
some places you have `this.` but in others you don't. Should they be the same?

Reply:
`this` is required if the method parameter name and the internal property are identical, optional otherwise

```kotlin
fun request(request: Request) = apply {
      this.request = request
}
```